src/components/checkbox/checkbox.spec.ts

```diff
@@ -103,6 +103,23 @@ describe('Checkbox', () => {
       );
     });
 
+    it('should correctly report validity status', async () => {
+      el = await fixture<IgcCheckboxComponent>(
+        html`<igc-checkbox required></igc-checkbox>`
+      );
+      expect(el.reportValidity()).to.equals(false);
+
+      el = await fixture<IgcCheckboxComponent>(
+        html`<igc-checkbox required checked></igc-checkbox>`
+      );
+      expect(el.reportValidity()).to.equals(true);
+
+      el = await fixture<IgcCheckboxComponent>(
+        html`<igc-checkbox></igc-checkbox>`
+      );
+      expect(el.reportValidity()).to.equals(true);
+    });
+
```

Comment:
We should add a test for igcChange on keyboard interaction (currently we test only on click - last test).

Reply:
Since browsers fire `click` after `keyup` (which should be called _activate_, imo) on Enter/Spacebar when the focus is on the input, this would be hard to emulate in the test environment with firing events programmatically.
We can assume the click handler tests cover this scenario until browsers drop the quirksmode (doubtful about that).